test/test_xpu.py

```diff
@@ -236,6 +237,39 @@ def convert_boolean_tensors(x):
 
             self.assertEqual(expect, actual)
 
+    def test_serialization_array_with_storage(self):
+        x = torch.randn(5, 5).xpu()
+        y = torch.IntTensor(2, 5).fill_(0).xpu()
+        q = [x, y, x, y.storage()]
```

Comment:
```suggestion
        q = [x, y, x, y.untyped_storage()]
```
This is going to raise a warning otherwsie

Reply:
If applied this change causes runtime error:
```
Traceback (most recent call last):
  File "/home/dvrogozh/git/pytorch/torch/testing/_internal/common_utils.py", line 2763, in wrapper
    method(*args, **kwargs)
  File "/home/dvrogozh/git/pytorch/torch/testing/_internal/common_device_type.py", line 432, in instantiated_test
    raise rte
  File "/home/dvrogozh/git/pytorch/torch/testing/_internal/common_device_type.py", line 419, in instantiated_test
    result = test(self, **param_kwargs)
  File "/home/dvrogozh/git/pytorch/test/test_xpu.py", line 245, in test_serialization_array_with_storage
    torch.save(q, f)
  File "/home/dvrogozh/git/pytorch/torch/serialization.py", line 613, in save
    _save(obj, opened_zipfile, pickle_module, pickle_protocol, _disable_byteorder_record)
  File "/home/dvrogozh/git/pytorch/torch/serialization.py", line 825, in _save
    pickler.dump(obj)
  File "/home/dvrogozh/git/pytorch/torch/serialization.py", line 803, in persistent_id
    raise RuntimeError(
RuntimeError: Cannot save multiple tensors or storages that view the same data as different types
```
Change was:
```
diff --git a/test/test_xpu.py b/test/test_xpu.py
index 2526b65e589..85212d919f9 100644
--- a/test/test_xpu.py
+++ b/test/test_xpu.py
@@ -239,8 +239,8 @@ if __name__ == "__main__":

     def test_serialization_array_with_storage(self):
         x = torch.randn(5, 5).xpu()
-        y = torch.IntTensor(2, 5).fill_(0).xpu()
-        q = [x, y, x, y.storage()]
+        y = torch.zeros(2, 5, dtype=torch.int, device="xpu")
+        q = [x, y, x, y.untyped_storage()]
         with tempfile.NamedTemporaryFile() as f:
             torch.save(q, f)
             f.seek(0)
```

Reply:
It seems this will need to be addressed after/during migration of tensor underlying storages to `untyped_storage()`. I left his unaddressed. If you have idea how to address this now - let me know.

Reply:
Ok, let's punt this to a follow up then!